src/platform/WebAppWayland.h

```diff
@@ -77,22 +58,18 @@ class WebAppWayland : public WebAppBase, WebPageBlinkObserver {
 		  int surface_id,
                   int width = 0, int height = 0,
                   int displayId = kUndefinedDisplayId,
-                  const std::string& location_hint = "",
-		  int surface_role = -1,
-		  int panel_type = -1);
+                  const std::string& location_hint = "", ShellSurface *surface = nullptr);
     WebAppWayland(const std::string& type, WebAppWaylandWindow* window,
                   int width = 0, int height = 0,
                   int displayId = kUndefinedDisplayId,
-                  const std::string& location_hint = "",
-		  int surface_role = -1,
-		  int panel_type = -1);
+                  const std::string& location_hint = "", ShellSurface *surface = nullptr);
```

Comment:
We should move away AGL code from WebAppWayland. We already have WebAppWaylandAGL, not sure why this needs to know about shell surface.